src/coreclr/jit/emitxarch.cpp

```diff
@@ -1529,20 +1506,20 @@ emitter::code_t emitter::AddRexXPrefix(const instrDesc* id, code_t code)
 {
     instruction ins = id->idIns();
 
-    if (UseEvexEncoding() && IsEvexEncodedInstruction(ins))
+    if (IsVexOrEvexEncodableInstruction(ins))
     {
-        if (TakesEvexPrefix(id))
+        if (TakesEvexPrefix(id) && codeEvexMigrationCheck(code)) // TODO-XArch-AVX512: Remove codeEvexMigrationCheck().
```

Comment:
We might actually be able to remove the `codeEvexMigrationCheck(code)` at this point, but I'm likewise leaving that to a future PR.